### PR TITLE
Convert to use the electron-installer-common ElectronInstaller class

### PR DESCRIPTION
**Pull request by malept**
Utilizes `electron-installer-common` 0.6.0.

### CODE REVIEW

File: src/installer.js [hunk 1]
```diff
@@ -18,164 +18,145 @@ const defaultRename = function (dest, src) {
...
-  const desktopSrc = options.desktopTemplate || path.resolve(__dirname, '../resources/desktop.ejs')
-  return common.createDesktop(options, dir, desktopSrc, 'BUILD')
-}
+  /**
+   * Get the hash of default options for the installer. Some come from the info
+   * read from `package.json`, and some are hardcoded.
+   */
+  generateDefaults () {
+    return common.readElectronVersion(this.userSupplied.src)
+      .then(electronVersion => Promise.all([common.readMetadata(this.userSupplied), redhatDependencies.forElectron(electronVersion, this.userSupplied.logger)]))
```
Comment (fcastilloec) [inline src/installer.js]: this lines is getting too long

File: src/installer.js [hunk 2]
```diff
@@ -184,31 +165,31 @@ module.exports = (data, callback) => {
...
-      data.logger(`Successfully created package at ${options.dest}`)
-      return options
+      data.logger(`Successfully created package at ${installer.options.dest}`)
+      return installer.options
     }).catch(err => {
       data.logger(common.errorMessage('creating package', err))
       throw err
     })
 
   return nodeify(promise, callback)
```
Comment (fcastilloec) [inline src/installer.js]: We need to remove callback support. It's too bad that only now I realize about this, which could have been part of a previous release

Comment (malept) [reply src/installer.js]: I think it's also in `-windows`.

Comment (malept) [reply src/installer.js]: I also have `nodeify` in `electron-installer-snap`. I was planning on getting rid of that dependency the next time there's a major version bump, for `-snap` hopefully that's when Node 6 is EOL'd.

**Comment by malept** [comment]
@fcastilloec FYI, I've made the requested changes. I'm probably just going to merge later today (here and in the other PRs) since you probably don't have time to look at the updates.

**Pull request merged**